agent.py

```diff
@@ -451,9 +408,31 @@ def _communicate_with_llm(self, prompt):
             messages=[{"role": "user", "content": prompt}],
             stream=True,
         )
-        return self._process_stream_until_complete(
-            current_stream, live_print_to_stdout=True
-        )
+        return self._process_stream_until_complete(current_stream, live_print_to_stdout=True)
+
+    def _call_open_api(self, params: str) -> str:
+        try:
+            params_dict = json.loads(params)
+            url = params_dict["url"]
+            method = params_dict.get("method", "GET").upper()
+            payload = params_dict.get("payload", None)
+            headers = params_dict.get("headers", {})
+
+            if method == "GET":
```

Comment:
I think it is better to have a switch-case statement here instead of continous if/else. WDYT?